src/connectivity.jl

```diff
@@ -19,20 +19,26 @@ function connected_components! end
     T = eltype(g)
     nvg = nv(g)
 
-    visitor = LightGraphs.ComponentVisitorVector(label, zero(T))
-    colormap = fill(0, nvg)
-    queue = Vector{T}()
-    sizehint!(queue, nvg)
-    for v in vertices(g)
-        if label[v] == zero(T)
-            visitor.labels[v] = v
-            visitor.seed = v
-            traverse_graph!(g, BreadthFirst(), v, visitor; vertexcolormap=colormap, queue=queue)
+    for u in vertices(g)
+        label[u] != zero(T) && continue
+        label[u] = u
+        Q = Vector{T}()
+        push!(Q, u)
+        while !isempty(Q)
+            src = shift!(Q)
+            vertexneighbors = out_neighbors(g, src)
+            for vertex in vertexneighbors
```

Comment:
well, what we're going to do here is generalize these functions where it makes sense with `dir={:out, :in}` (in accordance with your suggestion below), so we will have `for vertex in neighfn(g, src)` instead. But yes.